### PR TITLE
[Runtime] calculate accuracy of runtime

### PR DESCRIPTION
**Pull request by zhenwei-intel**
## Type of Change

feature

## Description

usage:

```python
python intel_extension_for_transformers/llm/runtime/graph/scripts/test_acc.py --model_name ~/lzw/Llama-2-7b-chat-hf/ --model_format torch --tasks piqa
```

outputs:
```
|     Task     |Version|Metric|Value |   |Stderr|
|--------------|------:|------|-----:|---|-----:|
|lambada_openai|      0|ppl   |3.6585|±  |0.0967|
|              |       |acc   |0.6839|±  |0.0065|

Accuracy for lambada_openai is: 0.6838734717640209
```

Meta-llama/Llama-2-7b-chat-hf on lamabda_openai:

|  | acc | ppl |
| ---- | ---- | ---- |
| Fp32<br> | 0.6986 | 3.3715 |
| Q4j | 0.6839 | 3.6585 |
| GPTQ | 0.6953 | 3.5189 |

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by airMeng** [comment]
why here not neural_speed repo

**Comment by zhenwei-intel** [comment]
> why here not neural_speed repo

customer need this function, which will be cherry-picked to ns

File: intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py [hunk 1]
```diff
@@ -83,6 +83,10 @@ def main(args_in: Optional[List[str]] = None) -> None:
 
     f.write(struct.pack("f", config["rms_norm_eps"]))
     f.write(struct.pack("f", config["rope_theta"] if "rope_theta" in config else 10000))
+    rope_scale = 1
+    if "rope_scaling" in config and config["rope_scaling"] is not None:
+        rope_scale = config["rope_scaling"]["factor"] if "factor" in config["rope_scaling"] else 1
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:88]: implementation quite confusing
```suggestion
rope_scale = config.get("rope_scaling", {}).get("factor", 1)
```

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:88]:
```
llama/Llama-2-7b-chat-hf/config.json                                                                                                                              
19:  "rope_scaling": null,

Magicoder-S-DS-6.7B/config.json                                                                                                                                   
21:  "rope_scaling": { 
  "factor": 4.0,
```
other models do not have this parameter

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:88]: `config.get("rope_scaling", {})` will return None, None can not `.get("factor", 1)`

Comment (airMeng) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:88]: > `config.get("rope_scaling", {})` will return None, None can not `.get("factor", 1)`

I just tried, anything missing?
```
(base) hengyume@DUT7050PVC:~/neural-speed/build> python
Python 3.9.18 (main, Sep 11 2023, 13:41:44) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> {}.get("a", 1)
1
>>> {"a": 1}.get("b", {}).get("c", 1)
1
```

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:88]: > "rope_scaling": null,

`config.get("rope_scaling", {})` will return None, not {}

Comment (airMeng) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:88]: get it, how about 
```suggestion
    rope_scale = 1
    if config.get("rope_scaling") is not None:
        rope_scale = config["rope_scaling"].get("factor", 1)
```

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:88]: great! 
need change to one line?
`rope_scale = config["rope_scaling"].get("factor", 1) if config.get("rope_scaling") is not None else 1`

Comment (airMeng) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py:88]: one-line seems complicated

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptq_mistral.py]: updated

File: intel_extension_for_transformers/llm/runtime/graph/scripts/test_acc.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    tasks = args.tasks
+    results = evaluate(
+        model="hf-causal",
+        model_args=f'pretrained="{model_name}"',
+        tasks=[f"tasks"],
+        # limit=5,
+        model_format=f"{model_format}"
+    )
+
+    print(results)
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/runtime/graph/scripts/test_acc.py:36]: /no new line

**Review by airMeng — APPROVED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/scripts/test_acc.py (hunk 2 above)
Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/test_acc.py]: fixed

**Review by a32543254 — APPROVED** [review]
LGTM

**Comment by VincyZhang** [comment]
NeuralChat UT failure is irrelevant.

**Review by VincyZhang — APPROVED** [review]

**Pull request merged**